src/components/WalletStatementModal/index.native.js

```diff
@@ -1,79 +1,76 @@
-import React from 'react';
+import React, {useCallback, useRef} from 'react';
 import {WebView} from 'react-native-webview';
 import lodashGet from 'lodash/get';
 import {withOnyx} from 'react-native-onyx';
 import _ from 'underscore';
-import withLocalize from '../withLocalize';
+import {useNavigation} from '@react-navigation/native';
 import ONYXKEYS from '../../ONYXKEYS';
-import compose from '../../libs/compose';
 import {walletStatementPropTypes, walletStatementDefaultProps} from './WalletStatementModalPropTypes';
 import FullScreenLoadingIndicator from '../FullscreenLoadingIndicator';
 import * as Report from '../../libs/actions/Report';
-import Navigation from '../../libs/Navigation/Navigation';
 import ROUTES from '../../ROUTES';
 
-class WalletStatementModal extends React.Component {
-    constructor(props) {
-        super(props);
+const IOU_ROUTES = [ROUTES.IOU_REQUEST, ROUTES.IOU_SEND];
+const renderLoading = () => <FullScreenLoadingIndicator />;
 
-        this.authToken = lodashGet(props, 'session.authToken', null);
-        this.navigate = this.navigate.bind(this);
-    }
+function WalletStatementModal({statementPageURL, session}) {
+    const webViewRef = useRef();
+    const authToken = lodashGet(session, 'authToken', null);
+    const {navigate} = useNavigation();
 
     /**
      * Handles in-app navigation for webview links
      *
      * @param {String} params.type
      * @param {String} params.url
      */
-    navigate({type, url}) {
-        if (!this.webview || (type !== 'STATEMENT_NAVIGATE' && type !== 'CONCIERGE_NAVIGATE')) {
-            return;
-        }
+    const handleNavigationStateChange = useCallback(
+        ({type, url}) => {
+            if (!webViewRef.current || (type !== 'STATEMENT_NAVIGATE' && type !== 'CONCIERGE_NAVIGATE')) {
+                return;
+            }
+
+            if (type === 'CONCIERGE_NAVIGATE') {
+                webViewRef.current.stopLoading();
+                Report.navigateToConciergeChat();
+            }
 
-        if (type === 'CONCIERGE_NAVIGATE') {
-            this.webview.stopLoading();
-            Report.navigateToConciergeChat();
-        }
+            if (type === 'STATEMENT_NAVIGATE' && url) {
```

Comment:
Can we consider in a scope of this PR to extract `STATEMENT_NAVIGATE` and `CONCIERGE_NAVIGATE` to **CONST**? It looks like a type constants and used across other components.

Reply:
Good idea, it is used only in  wallet statement components right now. I've added a proposition, let me know if the naming is readable.